xblock/fields.py

```diff
@@ -828,31 +816,25 @@ def from_json(self, value):
         """
         Parse the date from an ISO-formatted date string, or None.
         """
-        if isinstance(value, basestring):
+        if value is None:
+            return None
 
+        if isinstance(value, basestring):
             # Parser interprets empty string as now by default
             if value == "":
                 return None
-
             try:
-                parsed_date = dateutil.parser.parse(value)
+                value = dateutil.parser.parse(value)
             except (TypeError, ValueError):
                 raise ValueError("Could not parse {} as a date".format(value))
 
-            if parsed_date.tzinfo is not None:  # pylint: disable=maybe-no-member
-                parsed_date.astimezone(pytz.utc)  # pylint: disable=maybe-no-member
-            else:
-                parsed_date = parsed_date.replace(tzinfo=pytz.utc)  # pylint: disable=maybe-no-member
-
-            return parsed_date
-
-        if value is None:
-            return None
-
-        if isinstance(value, datetime.datetime):
-            return value
+        if not isinstance(value, datetime.datetime):
+            raise TypeError("Value should be loaded from a string, not {}".format(type(value)))
```

Comment:
@smarnach slightly inaccurate warning: technically `from_json` will work just fine in three cases:

* value is None
* value is string that can be parsed as datetime
* value is a datetime already

So, it should probably say something like "value should be loaded from a string or datetime ...".

One more thing - I believe `from_json` should *only* work with json-serialized stuff, which is always a string. So, if it get's something different as input it is kind of wrong. - I believe it should still raise or at least log warning.